llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -1329,6 +1342,51 @@ struct PreservedModuleHashAnalysis
 
 AnalysisKey PreservedModuleHashAnalysis::Key;
 
+struct MachineInstrCounterAnalysis
+    : public AnalysisInfoMixin<MachineInstrCounterAnalysis> {
+  static AnalysisKey Key;
+
+  struct Result {
+    bool invalidate(MachineFunction &MF, const PreservedAnalyses &,
+                    MachineFunctionAnalysisManager::Invalidator &) {
+      return false;
```

Comment:
if this is never invalidated, then isn't `CountBefore` going to be wrong after the first pass that changes the instruction count?

Reply:
Oops, I mixed with another method by emitting them when this analysis is invalidated.😓

Reply:
I'm still not understanding how the current code works with multiple passes, it's still not getting invalidated.

But is it possible to simplify this by just having the instrumentation record the instruction count before as pass and compare that after the pass? and for a module pass, we can record every MachineFunction's count with a map. then there's no need to involve analyses/analysis managers